lib/active_merchant/billing/gateways/payment_solutions.rb

```diff
@@ -95,7 +95,7 @@ def add_customer_data(xml, payment, options)
           xml['d4p1'].LastName payment.last_name
           xml['d4p1'].Phone address[:phone] if address[:phone]
           xml['d4p1'].PostalCode address[:zip] if address[:zip]
-          xml['d4p1'].SendEmail false
+          xml['d4p1'].SendEmail address[:email].nil? ? false : true
```

Comment:
Good point, I had forgotten that `ActiveMerchant` uses `activesupport` ... so I have access to `.present?` and I did use it further down in that same implementation 🐼